src/uint256.h

```diff
@@ -91,6 +93,26 @@ class base_blob
     }
 };
 
+template <unsigned int BITS>
+consteval base_blob<BITS>::base_blob(std::string_view hex_str)
+{
+    // Non-lookup table version of HexDigit().
+    auto from_hex = [](const char c) -> int8_t {
+        if (c >= '0' && c <= '9') return c - '0';
+        if (c >= 'a' && c <= 'f') return c - 'a' + 0xA;
+        if (c >= 'A' && c <= 'F') return c - 'A' + 0xA;
+
+        assert(false); // Reached if ctor is called with an invalid hex digit.
```

Comment:
Might be a good idea to add a small suggestion in the developer-notes for using `throw` for these checks, since we already do so in https://github.com/bitcoin/bitcoin/blob/master/src/script/miniscript.h#L177 . I initially assumed this was not supported, but seems like that was fixed in recent compiler versions.

Reply:
Iff I retouch, I might experiment with including the offending digit in the `throw`.

Reply:
Yeah, I experimented with using `std::format` but at least GCC would not print the actual thrown value anyway.